components-api/src/main/scala/pl/touk/nussknacker/engine/api/typed/TypeConversionHandler.scala

```diff
@@ -47,8 +71,9 @@ object TypeConversionHandler {
     }
   }
 
-  private def handleStringToValueClassConversions(givenClass: TypedClass, superclassCandidate: TypedClass): Boolean = {
-    givenClass == Typed[String] && ValueClassesThatBeConvertedFromString.exists(ClassUtils.isAssignable(superclassCandidate.klass, _, true))
+  private def handleStringToValueClassConversions(givenType: SingleTypingResult, superclassCandidate: TypedClass): Boolean = {
+    givenType.isInstanceOf[TypedObjectWithValue] &&
```

Comment:
Let's replace this with pattern matching so it won't be possible to desynchronize the code of  `isInstanceOf` and `asInstanceOf`?